src/index.js

```diff
@@ -3,22 +3,33 @@ import queryString from 'query-string';
 const defaultUrlRegExp = /^(\w+:\/\/[^/?]+)?(.*?)(\?.+)?$/;
 const protocolRelativeUrlRegExp = /^(\/\/[^/?]+)(.*?)(\?.+)?$/;
 
-function splitUrl(partsStr, { protocolRelative }) {
+function parseUrl(partsStr, { protocolRelative }) {
     const match = (protocolRelative && partsStr.match(protocolRelativeUrlRegExp)) ||
                    partsStr.match(defaultUrlRegExp) ||
                   [];
 
-    const beforePathname = match[1] || '';
-    const pathname = (match[2] || '')
-    // Remove leading slashes
-    .replace(/^\/+/, '')
-    // Remove trailing slashes
-    .replace(/\/+$/, '')
-    // Normalize consecutive slashes to just one
-    .replace(/\/+/g, '/');
-    const afterPathname = (match[3] || '');
-
-    return { beforePathname, pathname, afterPathname };
+    const prefix = match[1] || '';
+
+    const getParts = (prePathname) =>
+        prePathname.split('/').filter((part) => part !== '');
+
+    const hasLeadingSlash = (prePathname) =>
+        prePathname.match(/^\/+/) !== null;
```

Comment:
You can use `regexp.test`, which is faster. See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test

Reply:
Fixed